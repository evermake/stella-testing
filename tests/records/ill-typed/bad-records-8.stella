language core;

extend with #records;

fn main(x : Nat) -> Nat {
  return
    { f = 0, g = fn(x : Bool) { return if x then succ(0) else 0 }}.f(false)
}
