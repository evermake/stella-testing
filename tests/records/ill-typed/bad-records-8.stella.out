TYPE ERROR:
Type Error Tag: [ERROR_NOT_A_FUNCTION]
expected a function type but got
  Nat
for the expression
  
{
  f = 0, g = fn (x : Bool)
  {
    return if x then succ (0) else 0
  }
}
. f
in the function call
  
{
  f = 0, g = fn (x : Bool)
  {
    return if x then succ (0) else 0
  }
}
. f (false)